core/src/main/kotlin/org/evomaster/core/problem/rest/service/RestResourceFitness.kt

```diff
@@ -58,43 +71,97 @@ class RestResourceFitness : AbstractRestFitness<RestIndividual>() {
+                    So 2 options: (1) always reset before calling SUT, or (2) always reset after call in which external
+                    setups were used.
+                 */
+                externalServiceHandler.resetServedRequests()
+
+                val externalServiceActions = a.getExternalServiceActions()
+
+                externalServiceActions.filterIsInstance<HttpExternalServiceAction>().forEach {
+                    // TODO: Handling WireMock for ExternalServiceActions should be generalised
+                    //  to facilitate other cases such as RPC and GraphQL
+                    it.buildResponse()
+                }
+
+                val restCallAction = a.getMainAction()
 
                 //TODO handling of inputVariables
-                registerNewAction(a, indexOfAction)
+                registerNewAction(restCallAction, indexOfAction)
 
                 val ok: Boolean
 
-                if (a is RestCallAction) {
-                    ok = handleRestCall(a, actionResults, chainState, cookies, tokens)
+                if (restCallAction is RestCallAction) {
+                    ok = handleRestCall(restCallAction, actionResults, chainState, cookies, tokens)
                     // update creation of resources regarding response status
                     val restActionResult = actionResults.filterIsInstance<RestCallResult>()[indexOfAction]
-                    call.getResourceNode().confirmFailureCreationByPost(call, a, restActionResult)
+                    call.getResourceNode().confirmFailureCreationByPost(call, restCallAction, restActionResult)
                     restActionResult.stopping = !ok
                 } else {
-                    throw IllegalStateException("Cannot handle: ${a.javaClass}")
+                    throw IllegalStateException("Cannot handle: ${restCallAction.javaClass}")
+                }
+
+                // get visited wiremock instances
+                val requestedExternalServiceUrls = externalServiceHandler.getRequestedExternalServiceUrls()
+                if(requestedExternalServiceUrls.isNotEmpty()) {
+                    fv.registerExternalServiceRequest(
+                        indexOfAction,
+                        requestedExternalServiceUrls
+                    )
+                }
+
+                // Mark the existing external service actions as used if there is call
+                //  made based on the absolute URL. Otherwise, mark it not used
+                requestedExternalServiceUrls.forEach { url ->
+                    externalServiceActions.filterIsInstance<HttpExternalServiceAction>().forEach { action ->
+                        if (action.request.absoluteURL == url) {
+                            action.confirmUsed()
+                        } else {
+                            // Code never reaches this part
+                            action.confirmNotUsed()
+                        }
+                        action.resetActive()
```

Comment:
might not `resetActive` here, otherwise the actual `active` info would be missing